### ISSUE TITLE
Allowing for multiple `testconf.py` with the same `parser.addoption` option

### ISSUE THREAD

**Issue by stas00** [pytest-dev/pytest#7974]
tldr: Currently, it's not possible to have 2 `testconf.py` files that both call  `parser.addoption` with the same custom option. Would it be possible to add an option for this command not to fail if it's called a second time or being able to check whether it has been loaded already?

The full story:

At [transformers](https://github.com/huggingface/transformers/) - there are 2 separate test suites with similar `testconf.py` each. 

Current layout:
```
tests/
    testconf.py
    test_*.py
examples/
    testconf.py
    test_*.py
```

Normally there is no issue there as they are always run separately. But there are 2 unusual circumstances where I encountered the 2 files collide.

1. `pytest-instafail` for some reason loads `tests/testconf.py` when I run the `examples` test suite. And `parser.addoption` crashes then as we have the same custom flag in each test suite.  So I can't use that otherwise handy plugin.
2. Recently I wanted to run several tests across these 2 test suites together as I was trying to detect some test interdependence - this didn't go well together either, again because  `parser.addoption` didn't like the same custom flag loaded twice.

The error is:
```
ValueError: option names {'--make_reports'} already added
```

To solve this I tried to see if I could check whether the option has been added already and then not call `parser.addoption` , but I don't see any API to do that. It requires hacking into the private API to gain that info and even then it's not so easy.
But probably there could be an optional argument to `parser.addoption` that allows multiple addition of the same? `ignore-duplicate=False` Or would you find this idea logically wrong? Then just being able to check that it was already added will work too.

I tried using the existing clargs as a workaround, but to no avail:

1. This one supposedly should not load `examples/testconf.py`, but it does:
```
pytest tests examples --confcutdir=examples
```
2. another suggestion I found was to do:
```
pytest tests examples -p no:examples/conftest.py
```
but it doesn't solve this either - both `conftest.py` files get loaded.

But even if these worked these solutions are cumbersome - it'd be much easier to solve the collision when it arises programmatically.

Thank you!

**Comment by nicoddemus** [comment]
> Currently, it's not possible to have 2 testconf.py files that both call parser.addoption with the same custom option. Would it be possible to add an option for this command not to fail if it's called a second time or being able to check whether it has been loaded already?

I believe the best solution would be to add a new function that returns if an option already exists... seems reasonable to have something like that regardless of your specific use case.

**Comment by stas00** [comment]
That works, thank you!

**Comment by RonnyPfannschmidt** [comment]
Why not have a shared pytest plugin for the shared parts?

**Comment by stas00** [comment]
It's a possibility too.
